resources/js/features/root/components/navbar/NavigationItem.tsx

```diff
@@ -0,0 +1,21 @@
+import { A, AnchorProps } from "@solidjs/router";
+import { IconTypes } from "solid-icons";
+import { Component, splitProps } from "solid-js";
+
+export interface NavigationItemProps extends AnchorProps {
+  icon: IconTypes;
+}
+
+export const NavigationItem: Component<NavigationItemProps> = (props) => {
+  const [local, rest] = splitProps(props, ["children", "icon"]);
+  return (
+    <A
+      {...rest}
+      class="mb-2 py-2 px-4 rounded-lg flex flex-row items-center gap-3 hover:bg-white"
+      activeClass="bg-white"
+    >
+      <local.icon size="25" />
```

Comment:
Utrata reaktywności. Ktoś może przekazać ikonę która się zmienia zależnie od czegoś tam, nie ma żadnego powodu na to nie pozwolić. Więc to powinien być Dynamic.

Reply:
sprawdzę czy tak faktycznie jest

Reply:
No ja to nawet sprawdziłem zanim napisałem :) I zastanawiam się z tego powodu czy wszystkie te miejsca gdzie masz props.Coś to nie powinno jednak być zmienione. Bo to jest słabość lintera że nie wyłapuje tej utraty reaktywności. No chyba że przyjmujemy sobie konwencję że komponenty przekazane do propsów z dużych liter muszą być static (wtedy local.Icon), ale to by trzeba ze sobą ustalić chyba.

Reply:
fakt, poprawiam na Dynamic